lib/puck/jar.rb

```diff
@@ -81,6 +81,11 @@ def initialize(configuration={})
     def create!
       FileUtils.mkdir_p(@configuration[:build_dir])
 
+      ["bin", "lib"].each do |directory|
+        if Dir.glob("#{directory}/{*,.*}").empty?
+          raise PuckError, "Cannot build Jar: #{directory} directory not present / empty"
+        end
+      end
```

Comment:
We should accept that `bin` doesn't exist, but `lib` should probably be required. So what we should do is to not assume that `bin` exists, as we do a bit further down, and just check for `lib` here.

There needs to be a test for this though.